packages/manager/apps/pci-databases-analytics/src/pages/services/[serviceId]/pools/_components/addEditConnectionPool.tsx

```diff
@@ -0,0 +1,342 @@
+import { useTranslation } from 'react-i18next';
+import { useParams } from 'react-router';
+import { useEffect } from 'react';
+
+import { Button } from '@/components/ui/button';
+import {
+  Dialog,
+  DialogClose,
+  DialogContent,
+  DialogDescription,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+} from '@/components/ui/dialog';
+import { Input } from '@/components/ui/input';
+import {
+  Form,
+  FormControl,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+} from '@/components/ui/form';
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from '@/components/ui/select';
+import { useToast } from '@/components/ui/use-toast';
+import SizeInput from './formPools/sizeSelect';
+
+import {
+  UseConnectionPoolFormProps,
+  useConnectionPoolForm,
+} from './formPools/formConnectionPool.hook';
+import {
+  useAddConnectionPool,
+  useEditConnectionPool,
+} from '@/hooks/api/connectionPool.api.hooks';
+import { ModalController } from '@/hooks/useModale';
+
+import { GenericUser } from '@/api/databases/users';
+import { database } from '@/models/database';
+import { POOL_CONFIG } from './formPools/connectionPool.const';
+import { ConnectionPoolEdition } from '@/models/databaseEdition';
+
+interface AddEditConnectionPoolModalProps {
+  isEdition: boolean;
+  editedConnectionPool?: database.postgresql.ConnectionPool;
+  connectionPools?: database.postgresql.ConnectionPool[];
+  users: GenericUser[];
+  databases: database.service.Database[];
+  service: database.Service;
+  controller: ModalController;
+  onSuccess?: (connectionPool?: database.postgresql.ConnectionPool) => void;
+  onError?: (error: Error) => void;
+}
+const AddEditConnectionPoolModal = ({
+  isEdition,
+  editedConnectionPool,
+  connectionPools,
+  users,
+  databases,
+  service,
+  controller,
+  onSuccess,
+  onError,
+}: AddEditConnectionPoolModalProps) => {
+  const { projectId } = useParams();
+  const formProps: UseConnectionPoolFormProps = isEdition
+    ? { editedConnectionPool }
+    : { existingConnectionPools: connectionPools };
+
+  const { form } = useConnectionPoolForm(formProps);
```

Comment:
```suggestion
  const { form } = useConnectionPoolForm({
    editedConnectionPool,
    existingConnectionPools: connectionPools;
  });
```
I think that the form should adapt in case we are in edition mode, the parent component should not have specific logic here.